### ISSUE TITLE
pyproj now a dependency - OWSlib not pip installable on Windows anymore

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopython/OWSLib#285]
I noticed that since recently `pyproj` is added as a dependency: https://github.com/geopython/OWSLib/commit/d5d062bb57647c99e3e4e220496c44dce7fb7ab8 by @jachym

This has the effect that you cannot longer just do `pip install OWSLib` on a plain python installation on Windows, since pyproj is not a pure python packages but needs to be compiled (and only the source itself is on PyPI). 

I know there are other ways to install OWSLib (eg using conda), but could it be considered to have `pyproj` only as an optional dependency? So that pip installing it on Windows still works?

**Comment by GGummow** [comment]
I have the same concern but at an even deeper level as there is also no `pip3 install OWSLib` support with PyPi as is necessary for installing to a Python 3.x virtualenv

**Comment by GGummow** [comment]
Sorry, correction to above: There is also no `pip3 install python3-OWSLib`  support within PyPi as is necessary for installing to a Python 3.x virtualenv

**Comment by jorisvandenbossche** [comment]
@GGummow What do you exactly mean? OWSLib is python 3 compatible, you can install it with pip with no problem AFAIK. 
But any case, if you have a problem with it, maybe better to open a separate issue, as I raised this one to be specific about the `pyproj` dependency

**Comment by GGummow** [comment]
Please forgive me for my ignorance, my background is more algorithm development than ‘code cutting’ and I’m relatively new to Python & to Linux.

My understanding/experience is that I can successfully install OWSLib into a python 2.7 virtualenv from PyPI with `pip install OWSLib`, but if I try to do the same into a Python 3.4 virtualenv then I’m told that the file is not found (THIS IS WHERE ANY MISUNDERSTANDINGS I HAVE BEGAN) ; i.e. the code is notionally compatible but not totally transparently compatible.

To install into Ubuntu from Debian’s repository you can install OWSLib but there is a different install for Python 2<http://packages.ubuntu.com/wily/all/python-owslib/download> than there is from Python 3<http://packages.ubuntu.com/wily/all/python3-owslib/download>, perhaps this is just necessary to ensure that the destination is correct, I concluded that the libraries are in themselves different.

Please be blunt and tell me if I got that wrong and they are in fact the same owslib_0.9.0-1 file.

THANKS in advance

**Comment by jorisvandenbossche** [comment]
@GGummow if you do `pip install` it does not use the debian packages, but the ones on PyPI (https://pypi.python.org/pypi/OWSLib/0.9.2), and for this the same name (OWSLib) is used for both python 2 and 3. 
So normally, you should just be able to install it with `pip install OWSLib` (or `pip3` is that is the registered command). Did you try this? What is the error you got?

**Comment by tomkralidis** [comment]
@GGummow your questions are best put forth on the OWSLib-users [mailing list](https://geopython.github.io/OWSLib/#mailing-lists), as they are 1./ user questions and not bug related and 2./ not related to this ticket.

**Comment by GGummow** [comment]
I’m a newby, so sorry, understood, Joris Van den Bossche is sorting me out THANKS!

**Comment by GGummow** [comment]
(Python_3.x_virtualenv)$ pip install OWSLib       # fails
(Python_2.x_virtualenv)$ pip install OWSLib       # works

So having failed with PyPI (for py3.4 env) I turned to Debian packages as “plan B’
Had similar problems with pyscopg2 & setuptools, so maybe it’s my system configuration plus my ignorance.

**Comment by jorisvandenbossche** [comment]
@GGummow 'fails' is not very helpful. You will need to give at least the error message in order that we can help you further. 
But, as @tomkralidis asked, please do not comment further in this issue about that. Can you send the error you get to mailing list (as an answer to the mail you already send some days ago)

**Comment by GGummow** [comment]
How embarrassing, there was a problem with my virtualenv’s path.

Just tried to reproduce the error and its gone, fixing the virtualenv’s path probably fixed it.

I’m confused but I’m learning. RESOLVED

Many apologies and THANKS.

PS Awesome library!

**Comment by jorisvandenbossche** [comment]
@GGummow glad you found the problem

@tomkralidis do you have an opinion on my original issue I raised about `pyproj`?

**Comment by tomkralidis** [comment]
@jorisvandenbossche thanks for the info.  @jachym seems the pyproj depedency has caused packaging issues.  Aren't all these elements supposed to be 4326 anyway?

**Comment by tilmanb** [comment]
I installed successfully and can use owslib.wms with 64 bit Python 2.7 shipped by ArcGIS 10.6 using pyproj at version 1.9.6 .
I would suggest to install owslib on windows with
`c:\Python27\ArcGISx6410.6\Scripts\pip2.7.exe install pyproj==1.9.6 owslib`

This works for me and colleagues. I have not (and will not) run tests or anything with pyproj 1.9.6 against other parts of OWSLib, so that needs to be done before anyone updates docs or anything.
